site/en/docs/extensions/mv3/getstarted/tut-tabs-manager/index.md

```diff
@@ -0,0 +1,382 @@
+    "https://developer.chrome.com/docs/extensions/*",
+  ],
+});
+...
+```
+
+{% Details %}
+{% DetailsSummary %}
+💡 **Can I use Chrome APIs directly in the popup?**
+{% endDetailsSummary %}
+
+A popup and other extension pages can call any [Chrome API][doc-apis] because they are served from the
+chrome schema. For example `chrome-extension://EXTENSION_ID/popup.html`.
+
+{% endDetails %}
+
+#### Focus on a tab {: #focus}
+
+First, the extension will sort the list alphabetically. Then, when a list item is clicked, it will
+focus on that tab using `tabs.update()` and bring the window to the front using `windows.update()`.
+Add the following code to `popup.js`:
+
+{% Label %}popup.js:{% endLabel %}
+
+```js
+...
+const collator = new Intl.Collator();
+tabs.sort((a, b) => collator.compare(a.title, b.title));
+
+const template = document.getElementById("li_template");
+const elements = new Set();
+for (const tab of tabs) {
+  const element = template.content.firstElementChild.cloneNode(true);
+
+  const title = tab.title.split("-")[0].trim();
+  const pathname = new URL(tab.url).pathname.slice("/docs".length);
+
+  element.querySelector(".title").textContent = title;
+  element.querySelector(".pathname").textContent = pathname;
+  element.querySelector("a").addEventListener("click", async () => {
+    // need to focus window as well as the active tab
+    await chrome.tabs.update(tab.id, { active: true });
+    await chrome.windows.update(tab.windowId, { focused: true });
+  });
+
+  elements.add(element);
+}
+document.querySelector("ul").append(...elements);
+...
+```
+
+{% Details %}
+{% DetailsSummary %}
+💡 **Interesting JavaScript used in this code**
+{% endDetailsSummary %}
+
+- The [Collator](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator) is used to sort the tabs array by title in any language.
+- The [template tag](https://web.dev/webcomponents-template/) is used to
+  define an HTML element that can be cloned instead of using `document.createElement()` to create each item.
+- The [URL constructor](https://developer.mozilla.org/docs/Web/API/URL/URL) is used to create and parse URLs.
+
+{% endDetails %}
+
+#### Group the tabs {: #group}
+
+Add the following code to create a button that will group all the tabs using [`tabs.group()`]() and
+move them into the current window.
+
+{% Label %}popup.js:{% endLabel %}
+
+```js
+...
+const button = document.querySelector("button");
+button.addEventListener("click", async () => {
+  const group = await chrome.tabs.group({ tabIds: tabs.map(({ id }) => id) });
+  await chrome.tabGroups.update(group, { title: "DOCS" });
+});
+```
+
+The [TabGroups][api-tabgroups] API allows the extension to name the group and choose a background
+color. Add the `“tabGroups”` permission to the manifest by adding the following code:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "permissions": [
+    "tabGroups"
+  ]
+}
+```
```

Comment:
I'd suggest moving this above the previous code block. The intent with this change is to encourage developers to remember that they must request permission for an API before they can use that API in their code.

Reply:
Good idea! 👍